.github/workflows/engine-checks-optional.yml

```diff
@@ -103,6 +163,13 @@ jobs:
         run: ./run git-clean
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Download Engine Distribution
+        uses: actions/download-artifact@v4
+        with:
+          name: engine-distribution-macos-native,test
+      - run: |-
+          tar -xvf built-distribution.tar
+          rm built-distribution.tar
```

Comment:
- Nice way to transfer distribution bits between jobs
- the next command `./run backend test jvm` then runs jobs like:
   - `enso --run test/Base_Tests`, etc.?
- if so, then the only thing to address #12123 is to **duplicate each such execution** - e.g.:
   - `enso --jvm --run test/Base_Tests`
   - `enso --run test/Base_Tests`
- of course the above assumes the `built-distribution.jar` contains _native enso executable_ [built with some native option combination](https://github.com/enso-org/enso/blob/6dcb26f856c9b974fbefacc6d894e69d2d4d9ae5/docs/infrastructure/native-image.md#engine-runner-configuration) like `ENSO_LAUNCHER=native,test`